p2p-sync/src/main/java/org/smartregister/p2p/activity/P2pModeSelectActivity.java

```diff
@@ -0,0 +1,315 @@
+import org.smartregister.p2p.contract.P2pModeSelectContract;
+import org.smartregister.p2p.dialog.StartDiscoveringModeProgressDialog;
+import org.smartregister.p2p.dialog.StartReceiveModeProgressDialog;
+import org.smartregister.p2p.handler.OnActivityRequestPermissionHandler;
+import org.smartregister.p2p.handler.OnActivityResultHandler;
+import org.smartregister.p2p.handler.OnResumeHandler;
+import org.smartregister.p2p.interactor.P2pModeSelectInteractor;
+import org.smartregister.p2p.presenter.P2pModeSelectPresenter;
+import org.smartregister.p2p.util.Constants;
+import org.smartregister.p2p.util.DialogUtils;
+import org.smartregister.p2p.util.Permissions;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import timber.log.Timber;
+
+public class P2pModeSelectActivity extends AppCompatActivity implements P2pModeSelectContract.View {
+
+    private Button sendButton;
+    private Button receiveButton;
+
+    private P2pModeSelectContract.Presenter presenter;
+    private ArrayList<OnActivityResultHandler> onActivityResultHandlers = new ArrayList<>();
+    private ArrayList<OnResumeHandler> onResumeHandlers = new ArrayList<>();
+    private ArrayList<OnActivityRequestPermissionHandler> onActivityRequestPermissionHandlers = new ArrayList<>();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_p2p_mode_select);
+
+        sendButton = findViewById(R.id.btn_p2pModeSelectActivity_send);
+        receiveButton = findViewById(R.id.btn_p2pModeSelectActivity_receive);
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        initializePresenter();
+
+        sendButton.setOnClickListener(null);
+        receiveButton.setOnClickListener(null);
+
+        sendButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                presenter.onSendButtonClicked();
+            }
+        });
+
+        receiveButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                presenter.onReceiveButtonClicked();
+            }
+        });
+    }
+
+    @Override
+    public void enableSendReceiveButtons(boolean enable) {
+        sendButton.setEnabled(enable);
+        receiveButton.setEnabled(enable);
+    }
+
+    @Override
+    public void showReceiveProgressDialog(@NonNull DialogCancelCallback dialogCancelCallback) {
+        FragmentManager fragmentManager = getSupportFragmentManager();
+        StartReceiveModeProgressDialog newFragment = new StartReceiveModeProgressDialog();
+        newFragment.setDialogCancelCallback(dialogCancelCallback);
+
+        newFragment.show(fragmentManager, "dialog_start_receive_mode_progress");
+    }
+
+    @Override
+    public void showDiscoveringProgressDialog(@NonNull DialogCancelCallback dialogCancelCallback) {
```

Comment:
This method seems similar to `showReceiveProgressDialog` with only the message string resource different. Maybe it can be implemented as one method that one passes the message as parameter

Reply:
The dialogs are in different files because I thought that the final UIs on both dialogs will be different. In case we reuse a single dialog to show the other authentication and authorisation UI and all. Not sure? What do you think at this point?

Reply:
makes sense